route.go

```diff
@@ -6,9 +6,27 @@ import (
 	"sync/atomic"
 )
 
-type RouteConfig struct {
-	Matcher Matcher
-	Dests   []*Destination
+type RouteConfig interface {
+	Matcher() *Matcher
+	Dests() []*Destination
+}
+
+type baseRouteConfig struct {
+	matcher Matcher
+	dests   []*Destination
+}
+
+func (c baseRouteConfig) Matcher() *Matcher {
+	return &c.matcher
+}
+
+func (c baseRouteConfig) Dests() []*Destination {
+	return c.dests
+}
+
+type consistentHashingRouteConfig struct {
+	RouteConfig
```

Comment:
Er, yes, I did mean to embed baseRouteConfig. Apparently, though, [embedding interfaces is allowed](http://stackoverflow.com/questions/24537443/meaning-of-a-struct-with-embedded-anonymous-interface), and it just means I can initialize the field with anything implementing the interface. Nonetheless, I'll change this to baseRouteConfig to avoid confusion (and to do what I had originally intended).